Adding bubbles support

arlexTech opened the pull request:
Issue: https://github.com/nextcloud/talk-android/issues/3384

### 🖼️ Screenshots
<table>
  <tr>
    <td align="center"><b>General settings</b></td>
    <td align="center"><b>Conversation Info</b></td>
  </tr>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/14ae8f21-0738-42ed-8bc4-370febf75c37" width="300"></td>
    <td><img src="https://github.com/user-attachments/assets/ac5ec3d2-de8f-4268-8ba4-9ed4333c80f8" width="300"></td>
  </tr>
</table>

https://github.com/user-attachments/assets/d48935d7-b47d-4473-a7fc-c078683e4cc9

https://github.com/user-attachments/assets/0e90219a-8e0d-4319-9a36-71dc538aadac


### 🚧 TODO

- [x] Bubble Works
- [x] Activating bubbles in settings
- [x] Activating or not bubbles for each conversation
- [x] Use Conversation Avatar for bubble icon
- [x] Multiple bubbles for multiple conversations
- [x] Embedded view
- [x] Create a bubble notification upon received message

### 🏁 Checklist

- [ ] ⛑️ Tests (unit and/or integration) are included or not needed
- [ ] 🔖 Capability is checked or not needed 
- [ ] 🔙 Backport requests are created or not needed: `/backport to stable-xx.x`
- [ ] 📅 Milestone is set
- [ ] 🌸 PR title is meaningful (if it should be in the changelog: is it meaningful to users?)

mahibi commented:
thank you @arlexTech 
we will review soon!

mahibi commented on app/src/main/java/com/nextcloud/talk/chat/BubbleActivity.kt:
```diff
@@ -0,0 +1,70 @@
+    private fun openInMainApp() {
+        val intent = Intent(this, ChatActivity::class.java).apply {
+            putExtras(this@BubbleActivity.intent)
+            flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP
```
multiple instances of the app are opened when "open is app" is used.
https://developer.android.com/develop/ui/views/notifications/bubbles#launching-activities

mahibi commented on app/src/main/java/com/nextcloud/talk/jobs/NotificationWorker.kt:
```diff
@@ -546,10 +621,12 @@ class NotificationWorker(context: Context, workerParams: WorkerParameters) : Wor
+        )
             .setPriority(NotificationCompat.PRIORITY_HIGH)
             .setCategory(category)
-            .setLargeIcon(getLargeIcon())
```
why is this removed?

arlexTech commented:
For the back button in bubbled mode I replaced it by the talk icon in bubble mode which open the app in normal mode on the conversation list

> as `onBackPressed` is deprecated we want to avoid it and use `onBackPressedDispatcher`

fixed

> multiple instances of the app are opened when "open is app" is used. https://developer.android.com/develop/ui/views/notifications/bubbles#launching-activities

should be fixed, I couldn't open multiple anymore

>Please check the details at https://github.com/nextcloud/talk-android/security/code-scanning/92 which contains recommendations how to avoid this.
Using FLAG_IMMUTABLE was also done in other parts of the app.

I tried but didn't manage to make it work with FLAG_IMMUTABLE and found that bubbles need to be mutable (don't know if there is tricks to avoid it)
and the link is dead


> please try to extract the above code into a method

done

> why is this removed?

because it was preventing bubbles to show up but it should be fixed now



I wanted the handleOnBackPressed function to minimize the bubble but I tried for hours and never managed to make it work correctly, if anyone has an idea of how to do it, I'm open to try anything at this point

Tell me if i can do anything more to merge it

mahibi reviewed (changes requested):
Thanks to implement the suggestions @arlexTech 👍 We appreciate your efforts!

I commented on some things that must be changed.

There were changes on master in NotificationWorker that conflict with your branch. Could you resolve it?

Please also see the codacy warnings at
https://app.codacy.com/gh/nextcloud/talk-android/pull-requests/5554/issues

and sign your commits like described in
https://github.com/nextcloud/talk-android/pull/5554/checks?check_run_id=57331453438

If you need any help or nextcloud employees should take over, just let us know.

arlexTech commented:
I think I fixed everything you said but the emoji picker, for the emoji picker I tried but didn't managed to make it work, it simply never shows up

mahibi commented:
@arlexTech i'm currently taking a look and modify some things. Is it okay for you if i push to the branch?

arlexTech commented:
> Regarding:
> 
> > Could you please sync the fork and rebase on master to have the newest upstream changes?
> 
> @arlexTech let me know if you would like to have rights to the original repo. So you could create branches inside it if you would like to further contribute in the future and reviewing gets easier for nc devs..

Yes, since I use this app everyday I'm interested in developing it but I still need to learn a lot about this android app world

arlexTech commented:
Yes sorry, I don't have much time so if someone can help it would be great.
If not I'll try to work on it when I can because I really want this to be merged
